### ISSUE TITLE
Upgrade to cosmos-sdk 0.38

### ISSUE THREAD

**Issue by ethanfrey** [cosmwasm/wasmd#15]
## Summary

A new stable release should be out in the next few days. It's much nice to base on that, then on a moving master branch. It also allows other projects to re-use the `x/wasmd` module much easier.

## Proposal

- [ ] Pull in changes to github.com/cosmos/gaia since we forked off (That is, anything after `d6dfa141e2ae38a1ff9f53fca8078c0822671b95`, or about 20 commit).  
- [ ] Update the requirements to cosmos-sdk:0.38 (if not already done by the above)
- [ ] Update `x/wasmd` as needed for compatibilty

This should also include adding the upgrade module to the core app, which seems to have been done in https://github.com/cosmos/gaia/commit/982376964fdb2527e97ab84827757488721a044e

Please note that pulling in changes is a bit tricky as we are not a proper fork, but have changed all import paths (so `wasmd` can be used properly as a stand-alone program). This will likely involve more manual work than a simple merge and conflict clean-up (although you can try that first). I think cherry-picking commits one by one may end up the easier. Note that many commits seem minor requirement upgrades and doc improvements, so the number with conflicts should be relatively limited. Here is an idea:

```sh
git checkout -b upgrade
git remote add upstream https://github.com/cosmos/gaia
git fetch upstream
git merge upstream/master
```

but that produces 31 file conflicts, although 22 are in markdown files

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by sunnya97** [comment]
I've created a version that is upgraded to the latest version of the master branch.

https://github.com/sikkatech/wasmd/tree/sdk-bump-version

Will add as PR once upgraded to official `0.38` and after #6 is merged (is branched off of that).

**Comment by ethanfrey** [comment]
Thank you @sunnya97 :+1:

**Comment by ethanfrey** [comment]
@sunnya97 They are working on rc1 for `0.38`.

I merged a bunch in the meantime (esp. 0.6.3 cosmwasm with an overhaul on errors usage in x/wasmd).

It would be great if you could rebase on current master and do the update. Note, I don't just want to update the dep, but also merge in any relevant changes to `gaia`. In particular, they added the `upgrade` module, which I want, and seem to have done a number of other adjustments in app. We can remove many of the docs from gaia, but I want the app code to stay in sync as much as possible
